backend/controller/controller.go

```diff
@@ -200,7 +203,7 @@ type Service struct {
 	asyncCallsLock          sync.Mutex
 }
 
-func New(ctx context.Context, db *dal.DAL, config Config, runnerScaling scaling.RunnerScaling) (*Service, error) {
+func New(ctx context.Context, pool *pgxpool.Pool, db *dal.DAL, config Config, runnerScaling scaling.RunnerScaling) (*Service, error) {
```

Comment:
Let's only pass `Pool` in here - the `dal.DAL` was necessary before because it was being shared between different consumers, but now that we've split the DALs we can create it in this constructor.

Reply:
Perfect! I'll leave `dal` in the Service struct until I'm done refactoring out all the other packages.